modules/ethereum/src/lib.rs

```diff
@@ -835,6 +957,26 @@ impl<T: Trait<I>, I: Instance> Storage for BridgeStorage<T, I> {
 		// and now prune headers if we need to
 		self.prune_blocks(MAX_BLOCKS_TO_PRUNE_IN_SINGLE_IMPORT, finalized_number, prune_end);
 	}
+
+	fn blocks_by_number(&self, block_number: u64) -> Vec<Header> {
```

Comment:
If we get rid of `best_blocks()` this should not be needed, since the method is pretty heavy.